src/get-started/fwe/local-caching.md

```diff
@@ -0,0 +1,232 @@
+        _cachedData = await readData();
+    }
+    // Step 3: Return the value in the cache
+    return _cachedData!;
+}
+```
+
+There are many interesting ways to vary this strategy,
+including the location of the cache, the extent to which you
+preemptively write values to, or "warm", the cache; and others.
+
+## Common caching terminology
+
+Caching comes with its own terminology, some of which is
+defined and explained below.
+
+**Cache hit**
+: An app is said to have had a cache hit when the cache already
+  contained their desired information and loading it from the
+  real source of truth was unnecessary.
+
+**Cache miss**
+: An app is said to have had a cache miss when the cache was
+  empty and the desired data is loaded from the real source
+  of truth, and then saved to the cache for future reads.
+
+## Risks of caching data
+
+An app is said to have a **stale cache** when the data within 
+the source of truth has changed, which puts the app at risk
+of rendering old, outdated information.
+
+All caching strategies run the risk of holding onto stale data.
+Unfortunately, the action of verifying the freshness of a cache
+often takes as much time to complete as fully loading the data
+in question. This means that most apps tend to only benefit
+from caching data if they trust the data to be fresh at runtime
+without verification.
+
+To deal with this, most caching systems include a time limit
+on any individual piece of cached data. After this time limit
+is exceeded, would-be cache hits are treated as cache misses
+until fresh data is loaded.
+
+A popular joke among computer scientists is that "The two
+hardest things in computer science are cache invalidation,
+naming things, and off-by-one errors." 😄
+
+Despite the risks, almost every app in the world makes heavy
+use of data caching. The rest of this page explores multiple
+approaches to caching data in your Flutter app, but know that
+all of these approaches can be tweaked or combined for your
+situation.
+
+## Caching data in local memory
+
+The simplest and most performant caching strategy is an
+in-memory cache. The downside of this strategy is that,
+because the cache is only held in system memory, no data is
+retained beyond the session in which it is originally cached.
+(Of course, this "downside" also has the upside of automatically
+solving most stale cache problems!)
+
+Due to their simplicity, in-memory caches closely mimic
+the pseudocode seen above. That said, it is best to use proven
+design principles, like the [repository pattern][],
+to organize your code and prevent cache checks like the above
+from appearing all over your code base.
+
+Imagine a `UserRepository` class that is also tasked with
+caching users in memory to avoid duplicate network requests.
+Its implementation might look like this:
+
+```dart
+class UserRepository {
+  UserRepository(this.api);
+  
+  final Api api;
+  final Map<int, User?> _userCache = {};
+
+  Future<User?> loadUser(int id) async {
```

Comment:
I would use the early return to avoid nesting:

```
Future<User?> loadUser(int id) async {
  if (!_userCache.containsKey(id)) return _userCache[id];

  final response = await api.get(id);
  final user = response.statusCode == 200 ? User.fromJson(response.body) : null;

  return _userCache[id] = user;
}
```

Actually, It's not only about the structure of the code and nesting, but being very specific about the remaining code not being executed if the cache is hit.